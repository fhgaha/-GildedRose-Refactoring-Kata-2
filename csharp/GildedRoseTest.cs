using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using NUnit.Framework;
using System.Collections.Generic;
using System.Linq;

namespace csharp
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class GildedRoseTest
    {
        [Test]
        public void foo()
        {
            string[] names = new string[]
            {
                "foo",
                "Aged Brie",
                "Backstage passes to a TAFKAL80ETC concert",
                "Sulfuras, Hand of Ragnaros"
            };
            int[] sellIns = Enumerable.Range(-1, 15).ToArray();
            int[] qualities = new int[] { 0, 1, -1, 49, 50, 51 };

            CombinationApprovals.VerifyAllCombinations(DoStuff, names, sellIns, qualities);
        }

        private static string DoStuff(string name, int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new Item { Name = name, SellIn = sellIn, Quality = quality } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            string result = app.Items[0].ToString();
            return result;
        }

        [Test]
        public void Conjured()
        {
            Approvals.Verify(DoStuff("Conjured Pot", 10, 2));
        }
    }
}
