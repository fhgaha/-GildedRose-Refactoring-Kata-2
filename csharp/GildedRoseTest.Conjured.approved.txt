Conjured Pot, 9, 0